app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java

```diff
@@ -616,5 +663,75 @@ private PlayQueue getPlayQueue(final int index) {
         }
         return new SinglePlayQueue(streamInfoItems, index);
     }
+
+    private class RemoveWatchedStreams{
+        List<PlaylistStreamEntry> localItems = new ArrayList<>();
+        Long RemovedItemCount = 0l;
+        boolean thumbnailVideoRemoved = false;
+
+        // Do this in the main thread
+        protected void onPreExecute() {
+            showLoading();
+        }
+
+        // Do not do this in the main thread
+        protected Long doInBackground(List<PlaylistStreamEntry> playlist) {
+
+            HistoryRecordManager recordManager = new HistoryRecordManager(getContext());
+            Iterator<StreamHistoryEntry> it_history;
+            StreamHistoryEntry history_item;
+
+            Iterator<PlaylistStreamEntry> it_playlist = playlist.iterator();
+            PlaylistStreamEntry playlist_item = null;
+
+            boolean isNonDuplicate;
+
+            while (it_playlist.hasNext()) {
+                playlist_item = it_playlist.next();
+
+                it_history = recordManager.getStreamHistory().blockingFirst().iterator();
+
+                isNonDuplicate = true;
+                while (it_history.hasNext()) {
+                    history_item = it_history.next();
+                    if (history_item.streamId == playlist_item.streamId) {
+                        isNonDuplicate = false;
+                        break;
+                    }
+                }
+                if (isNonDuplicate) {
+                    localItems.add(playlist_item);
+                }
+                else
+                {
```

Comment:
Match the style of the repo: `} else {`
Also below: spaces between `if` and `(`: `if (`

Reply:
Fixed in commit 81bcac5